### PR TITLE
equalize epoch counts for EpochsTFR

### LINKED ISSUE #11776: add equalize_event_counts method to EpochsTFR
time-domain Epochs object already has this method so should be fairly straightforward.

requested here:
https://mne.discourse.group/t/equalize-epoch-counts-for-epochstfr/6879/2

### PR DESCRIPTION
**Pull request by CarinaFo**
fixed #11776

mne.epoch.Epochs.equalize_epoch_counts allows EpochsTFR as inputs

### CODE REVIEW

**Comment by CarinaFo** [comment]
Better to place the drop method for EpochsTFR in utils.GetEpochsMixin ?

File: mne/epochs.py
Comment (mscheltienne) [inline mne/epochs.py:3554]: I would avoid unclear variable names, `epoch` would be better than `e`.

**Comment by mscheltienne** [comment]
+1 to move the `drop` method to a mixin inherited by `BaseEpochs` and by `EpochsTFR`.

File: mne/epochs.py
Comment (drammock) [inline mne/epochs.py:3558]: you're changing this variable name from `event_times` to `event_indices`...  the original name was not accurate but the new name isn't either. Let's call it `sample_nums` (since the first column of `events` is the sample number)

```suggestion
    sample_nums = [epoch.events[:, 0] for epoch in epochs_list]
    indices = _get_drop_indices(sample_nums, method)
```

File: mne/epochs.py
Comment (drammock) [inline mne/epochs.py:3563]: ditto here; if you're going to change the parameter name let's change it to `sample_nums`.  But please also run `git grep _get_drop_indices` in your terminal to make sure that this file is the only place that the helper function is being called --- if not, we need to make sure that in the other places it's not being called with named parameters (if it is, then changing the parameter name here will cause those other usages to fail)

File: mne/time_frequency/tfr.py
Comment (drammock) [inline mne/time_frequency/tfr.py:2894]: @larsoner @cbrnr this is the bit we specifically wanted your input on: here @CarinaFo is defining `EpochsTFR.drop()` as a wrapper for `BaseEpochs.drop()`.  Are you OK with this approach, or prefer moving `BaseEpochs.drop()` to a mixin class that both `BaseEpochs` and `EpochsTFR` inherit from?

File: mne/epochs.py
Comment (drammock) [inline mne/epochs.py:3520]:
```suggestion
    """Equalize the number of trials in multiple Epochs or EpochsTFR instances.
```

File: mne/time_frequency/tfr.py
Comment (CarinaFo) [reply mne/time_frequency/tfr.py:2911]: So I assume the wrapper is okey? I would close the PR then?

**Comment by CarinaFo** [comment]
I added the PR #12207 to the changelog to make the test pass, not sure why the PR has two different issue numbers.

**Comment by drammock** [comment]
@larsoner this one is all green and ready for a second pair of eyes.

File: mne/tests/test_epochs.py
Comment (larsoner) [reply mne/tests/test_epochs.py:2858]:
```suggestion
```

File: mne/time_frequency/tfr.py
Comment (larsoner) [reply mne/time_frequency/tfr.py:2911]: In principle I think we usually use a mixin class for this but I think it's okay to do it this way for now

**Review by larsoner — APPROVED** [review]

**Comment by CarinaFo** [comment]
Should I wait for the auto-merge before I close the PR?

**Comment by drammock** [comment]
> Should I wait for the auto-merge before I close the PR?

Don't close the PR. It will automatically close when it's merged... But there are 2 failing tests that are preventing auto merge. I'll look tomorrow, they may be unrelated to this PR (and thus you wouldn't need to fix them here) but let's be sure

**Pull request merged**
